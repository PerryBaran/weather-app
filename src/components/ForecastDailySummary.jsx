import React, { useContext } from "react";
import PropTypes from "prop-types";
import WeatherIcon from "react-icons-weather";
import "../styles/ForecastDailySummary.css";
import UnitContext from "../context/UnitContext";
import getUnits from "../helpers/getUnits";

function ForecastDailySummary({
  dateTime,
  date,
  weather,
  icon,
  temp,
  handleForecastSelect,
}) {
  const selectedUnits = useContext(UnitContext);
  const { tempUnits } = getUnits(selectedUnits);
  const description = weather ? weather.split(" ") : [weather];

  return (
    <div
      className="forecast-daily-summary"
      data-testid="forecast-daily-summary"
    >
      <p className="forecast-daily-summary__date">{date}</p>
      <div className="forecast-daily-summary__icon" data-testid="forecast-icon">
        <WeatherIcon name="owm" iconId={icon} />{" "}
      </div>
      <p className="forecast-daily-summary__description">
        {description[0]}
        {description.length === 2 ? <span>{description[1]}</span> : null}
      </p>
      <p className="forecast-daily-summary__temp">
        {temp ? `${temp}${tempUnits}` : null}
      </p>
      <button
        type="button"
        onClick={() => handleForecastSelect(dateTime)}
        className="forecast-daily-summary__more-details"
      >
        More Details
      </button>
    </div>
  );
}

ForecastDailySummary.propTypes = {
  dateTime: PropTypes.number.isRequired,
  date: PropTypes.string.isRequired,
  weather: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
  temp: PropTypes.string.isRequired,
  handleForecastSelect: PropTypes.func.isRequired,
};

export default ForecastDailySummary;
