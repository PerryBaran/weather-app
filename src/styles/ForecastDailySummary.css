.forecast-daily-summary {
  text-align: center;
  display: flex;
  flex-direction: column;
  justify-items: center;
  align-items: center;
  gap: 10px;
  border-left: 4px solid var(--light-blue);
  border-bottom: 4px solid var(--light-blue);
  width: 100%;
  padding-top: 5px;
}

.forecast-daily-summary:first-child {
  border-left: none;
}

.forecast-daily-summary__date {
  font-size: 1rem;
}

.forecast-daily-summary__icon {
  font-size: min(4rem, 5vw);
}

.forecast-daily-summary__description {
  font-size: min(1rem, 2.5vw);
}

.forecast-daily-summary__temp {
  font-size: min(1rem, 2.5vw);
}

.forecast-daily-summary__more-details {
  all: unset;
  width: 100%;
  padding: 5px 0;
  border-top: 4px solid var(--light-blue);
}

@media screen and (max-width: 780px) {
  .forecast-daily-summary__date {
    width: 50px;
  }
}