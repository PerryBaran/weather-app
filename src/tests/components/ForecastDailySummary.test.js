import React from "react";
import { render } from "@testing-library/react";
import ForecastDailySummary from "../../components/ForecastDailySummary";

describe("ForecastDailySummary", () => {
  const validProps = {
    dateTime: 20,
    date: "today",
    weather: "cold",
    icon: "800",
    temp: "20",
    handleForecastSelect: () => {},
  };

  describe("snapshot", () => {
    it("renders corretly", () => {
      const { asFragment } = render(
        <ForecastDailySummary
          dateTime={validProps.dateTime}
          date={validProps.date}
          weather={validProps.weather}
          icon={validProps.icon}
          temp={validProps.temp}
          handleForecastSelect={validProps.handleForecastSelect}
        />
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe("props", () => {
    it("renders correct values for props", () => {
      const { getByText, getByTestId } = render(
        <ForecastDailySummary
          dateTime={validProps.dateTime}
          date={validProps.date}
          weather={validProps.weather}
          icon={validProps.icon}
          temp={validProps.temp}
          handleForecastSelect={validProps.handleForecastSelect}
        />
      );

      expect(getByText(validProps.date)).toHaveAttribute(
        "class",
        "forecast-daily-summary__date"
      );
      expect(getByText(validProps.weather).closest("div")).toHaveAttribute(
        "class",
        "forecast-daily-summary__description"
      );
      expect(getByTestId("forecast-icon")).toHaveAttribute(
        "class",
        "forecast-daily-summary__icon"
      );
      expect(getByText(`${validProps.temp}Â°C`)).toHaveAttribute(
        "class",
        "forecast-daily-summary__temp"
      );
    });
  });
});
