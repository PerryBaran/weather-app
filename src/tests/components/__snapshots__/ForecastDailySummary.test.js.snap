// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDailySummary snapshot renders corretly 1`] = `
<DocumentFragment>
  <div
    class="forecast-daily-summary"
    data-testid="forecast-daily-summary"
  >
    <p
      class="forecast-daily-summary__date"
    >
      Thu 
      <br />
       Jan 01
    </p>
    <div
      class="forecast-daily-summary__icon"
      data-testid="forecast-icon"
    >
      <i
        class="wi wi-day-sunny"
        name="owm"
      />
       
    </div>
    <p
      class="forecast-daily-summary__description"
    >
      cold
    </p>
    <p
      class="forecast-daily-summary__temp"
    >
      20K
    </p>
    <button
      class="forecast-daily-summary__more-details"
      type="button"
    >
      More Details
    </button>
  </div>
</DocumentFragment>
`;
